Authentication using OpenKeystore SSH API

vexofp opened the pull request:
Implementation for #71 . It seems to work well with my Yubikey 5 over NFC.

Integrating the OpenKeychain API's model (lots of PendingIntents) with the existing code (mostly modal dialog boxes) was a bit interesting. The result is not quite as elegant as I would like, but manages to avoid any intrusive refactoring of the existing authentication methods.

razic commented:
@vexofp very nice

MSF-Jarvis commented on app/src/main/java/com/zeapo/pwdstore/git/GitActivity.java:
```diff
@@ -676,23 +681,62 @@ protected void onActivityResult(int requestCode, int resultCode,
         }
     }
 
+    @Override
```
Useless override? You're not doing anything here

vexofp replied:
Yep, this was left-over from debugging.

zeapo commented:
@MSF-Jarvis we have only few tests ;(

I can start looking into this. For the moment, this needs manual testing

zeapo commented:
Tested manually. It works fine. Openkeychain is surprisingly simple to use for these kind of use cases :)

MSF-Jarvis commented:
> Tested manually. It works fine. Openkeychain is surprisingly simple to use for these kind of use cases :)

Sweet, I'll rebase and merge this.

zeapo commented:
@vexofp I get asked to select the key on every call to jgit. Is it possible to make that selection when the user opt for OpenKeychain authentication?

Also, can you add an item in the FAQ on how to create an Authentication subkey in OpenKeychain?

razic commented:
I tested this feature manually as well. It works great. Thank you very much @vexofp. To me personally, this was the feature I was missing the absolute most.

@zeapo's suggestion with regard to saving the preferred key such that you are not prompted on every git operation would definitely be a good improvement to have.

Thanks again! Looking forward to seeing this merged.

bkchr commented:
Works for me as well :) Would be nice to have this into official releases!

vexofp commented:
I believe I have addressed the last few review comments and added a section to the [wiki](https://github.com/zeapo/Android-Password-Store/wiki/Generate-SSH-Key#from-openkeychain-coming-soon) on generating authentication keys in OpenKeychain, as requested by @zeapo.

I can try to work on saving the chosen key ID when I have some time, but that might be better as a separate PR.

bkchr commented:
Nice :) Hopefully this gets merged then :)

zeapo commented:
I'll test this tomorrow and then merge it.

Le mar. 2 avr. 2019 à 11:22, Bastian Köcher <notifications@github.com> a
écrit :

> Nice :) Hopefully this gets merged then :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zeapo/Android-Password-Store/pull/486#issuecomment-478915241>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAQgBqhA4i_8IFcr_fEpbYJ86ZoLOihvks5vcyFDgaJpZM4bO3k4>
> .
>

therealpxc commented:
What an exciting feature!

zeapo commented:
thanks :)
